packages/zosfiles/src/methods/upload/Upload.ts

```diff
@@ -348,19 +348,16 @@ export class Upload {
                 // If dsnameIndex === -1, it means we could not find the given data set.
                 // We will attempt the upload anyways so that we can forward/throw the proper error from z/OS MF
                 const dsInfo = listResponse.apiResponse.items[dsnameIndex];
-                switch (dsInfo.dsorg) {
-                    case "PO":
-                    case "PO-E":
-                        isUploadToPds = true;
-                        break;
-                    default:
+                if(dsInfo.dsorg.startsWith("PO")) {
```

Comment:
`dsorg` is not a required property when listing data sets - we can add optional chaining to this check to prevent a TypeError:

```suggestion
                if(dsInfo.dsorg?.startsWith("PO")) {
```